metrics-core/src/main/java/com/codahale/metrics/ScheduledReporter.java

```diff
@@ -145,6 +145,26 @@ public void run() {
     }
 
     /**
+     * Starts the reporter polling at the given period.
+     *
+     * @param period       the amount of time between polls
+     * @param initialDelay the time to delay the first execution
+     * @param unit         the unit for {@code period}
+     */
+    public void start(long period, long initialDelay, TimeUnit unit) {
+       executor.scheduleAtFixedRate(new Runnable() {
```

Comment:
I think we could avoid duplication and implement the method as:

```java
synchronized public void start(long initialDelay, long period, TimeUnit unit) {
        if (this.scheduledFuture != null) {
            throw new IllegalArgumentException("Reporter already started");
        }
        this.scheduledFuture = executor.scheduleAtFixedRate(new Runnable() {
           @Override
           public void run() {
               try {
                   report();
               } catch (RuntimeException ex) {
                   LOG.error("RuntimeException thrown from {}#report. Exception was suppressed.", ScheduledReporter.this.getClass().getSimpleName(), ex);
               }
           }
       }, initialDelay, period, unit);
   }
```

as then make the `start(long period, TimeUnit unit)` method to delegate to it as
```java
start(period, period, unit)
```